Elements/src/Ceiling.cs

```diff
@@ -0,0 +1,188 @@
+        /// <param name="name">The name of the ceiling.</param>
+        public Ceiling(Polygon perimeter,
+                      double thickness,
+                      Material material = null,
+                      Transform transform = null,
+                      Representation representation = null,
+                      bool isElementDefinition = false,
+                      Guid id = default(Guid),
+                      string name = null) : base(transform != null ? transform : new Transform(),
+                                                 material != null ? material : BuiltInMaterials.Concrete,
+                                                 representation != null ? representation : new Representation(new List<SolidOperation>()),
+                                                 isElementDefinition,
+                                                 id != default(Guid) ? id : Guid.NewGuid(),
+                                                 name)
+        {
+            if (thickness <= 0.0)
+            {
+                throw new ArgumentOutOfRangeException("The ceiling could not be created. The thickness of the ceiling must be greater than 0.0.");
+            }
+
+            if (!perimeter.Normal().IsParallelTo(Vector3.ZAxis))
+            {
+                throw new ArgumentOutOfRangeException("The ceiling could not be created. The perimeter polygon must lie on XY plane");
+            }
+
+            // we do not need null check cause id there is no vertices it will fail on calculating Normal
+            this.Elevation = perimeter.Vertices.First().Z;
+            this.Perimeter = perimeter.Project(new Plane(Vector3.Origin, Vector3.ZAxis));
+            this.Thickness = thickness;
+            Transform.Move(Vector3.ZAxis * Elevation);
+        }
+
+        /// <summary>
+        /// The Profile of the Ceiling computed from its Perimeter and the Openings.
+        /// </summary>
+        /// <returns></returns>
+        public Profile GetProfile()
+        {
+            return new Profile(Perimeter, Openings?.Select(o => o.Perimeter).ToList());
+        }
+
+        /// <summary>
+        /// Add an Opening to the Ceiling.
+        /// </summary>
+        /// <param name="perimeter">The plan perimeter of the ceiling. It should lie on XY plane.
+        /// The Z coordinate will be ignored</param>
+        /// <param name="depthFront"></param>
+        /// <param name="depthBack"></param>
+        public void AddOpening(Polygon perimeter, double depthFront = 1, double depthBack = 1)
+        {
+            if (!perimeter.Normal().IsParallelTo(Vector3.ZAxis))
+            {
+                throw new ArgumentOutOfRangeException("The opening could not be created. The perimeter polygon must lie on XY plane");
+            }
+
+            var opening = new Opening(perimeter.Project(new Plane(Vector3.Origin, Vector3.ZAxis)), perimeter.Normal(), depthFront, depthBack);
```

Comment:
Will this work? If we project the opening down to the XY plane, and depthfront/back is set to 1, then the resulting opening won't cut the ceiling. I do like this API taking care of creating the opening for the user though. Could we adjust the opening's transform up to the elevation of the ceiling?

Reply:
I removed  depthfront/back for now. We are using the only wall thickness at the moment as depthFront. I added a comment that we will need to be able to use opening with custom depthFront/depthBack, but we don't need it at the moment. Everything is kept with Z = 0, we keep elevation and additional transform, if needed.